specifyweb/specify/migrations/0002_geo.py

```diff
@@ -33,63 +32,75 @@ class Migration(migrations.Migration):
     ]
 
     def create_default_collection_types(apps, schema_editor):
-        # Create default collection types for each discipline
-        for discipline in Discipline.objects.all():
+        # Create default collection types for each collection, named after the discipline
+        for collection in Collection.objects.all():
+            discipline = collection.discipline
             discipline_name = discipline.name
-            if CollectionObjectType.objects.filter(name=discipline_name).exists():
-                continue
-            cot = CollectionObjectType.objects.create(
+            cot, created = CollectionObjectType.objects.get_or_create(
                 name=discipline_name,
-                isdefault=True,
-                collection = Collection.objects.filter(discipline=discipline).first(),
-                taxontreedef=discipline.taxontreedef,
+                collection=collection,
+                taxontreedef=discipline.taxontreedef
             )
-            cot.save()
+            # Update CollectionObjects' collectionobjecttype for the discipline
+            Collectionobject.objects.filter(collection=collection).update(collectionobjecttype=cot)
+            collection.collectionobjecttype = cot
+            collection.save()
 
-        # Iteratively update CollectionObjects' collectionobjecttype for each discipline in a bulk operation
-        for discipline in Discipline.objects.all():
-            discipline_name = discipline.name
-            cot = CollectionObjectType.objects.get(name=discipline_name)
-            (Collectionobject.objects
-                .filter(collection__discipline=discipline)
-                .update(cotype=cot))
+    def revert_default_collection_types(apps, schema_editor):
+        # Reverse handeled by table deletion.
+        pass
```

Comment:
I think i'm missing something here, but how/where is this handled by table deletion? 
Do you mean we are marking this part of the migration as 'reversible' but doing nothing here and that the changes will be 'reversed' when the CollectionObjectTypes are deleted? 

If this is the case, then ignore this comment (assuming that Django deletes the created CollectionObjectTypes and sets the `collectionobjecttype` on the `Collection` records to null when reverting the other operations in the migration).

Reply:
Not having a reverse function works for operations like `migrations.CreateModel` and `migrations.AddField`. But `migrations.RunPython` needs to explicitly have a reverse function, or Django assumes it is not reversible.